internal/resource/vault_test.go

```diff
@@ -0,0 +1,232 @@
+package resource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	model "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	"github.com/kong/koko/internal/model/json/validation"
+	"github.com/kong/koko/internal/test/util"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestNewVault(t *testing.T) {
+	r := NewVault()
+	require.NotNil(t, r)
+	require.NotNil(t, r.Vault)
+}
+
+func TestVault_Type(t *testing.T) {
+	require.Equal(t, TypeVault, NewVault().Type())
+}
+
+func TestVault_ProcessDefaults(t *testing.T) {
+	vault := NewVault()
+	require.Nil(t, vault.ProcessDefaults(context.Background()))
```

Comment:
nit: `require.NoError` is better suited here.

Reply:
fixed

Reply:
@GGabriele these are included.